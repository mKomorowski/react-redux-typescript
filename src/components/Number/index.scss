.number {
  &.positive {
    color: green;
  }

  &.negative {
    color: red;
  }
}
