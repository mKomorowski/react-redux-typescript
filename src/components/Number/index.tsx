import React, { FC } from "react";
import classnames from "classnames";

import "./index.scss";

type NumberProps = {
  value: number;
};

const Number: FC<NumberProps> = ({ value }) => (
  <span
    className={classnames("number", {
      positive: value > 0,
      negative: value < 0,
    })}
  >
    {value}
  </span>
);

export default Number;
