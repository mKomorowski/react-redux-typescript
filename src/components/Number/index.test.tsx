import React from "react";
import { render } from "@testing-library/react";

import Number from "./index";

describe("Number component", () => {
  it("renders correctly for '0'", () => {
    const { container } = render(<Number value={0} />);

    expect(container).toMatchSnapshot();
  });

  it("renders correctly for positive number", () => {
    const { container } = render(<Number value={1} />);

    expect(container).toMatchSnapshot();
  });

  it("renders correctly for negative number", () => {
    const { container } = render(<Number value={-1} />);

    expect(container).toMatchSnapshot();
  });
});
