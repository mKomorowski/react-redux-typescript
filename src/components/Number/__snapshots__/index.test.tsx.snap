// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Number component renders correctly for '0' 1`] = `
<div>
  <span
    class="number"
  >
    0
  </span>
</div>
`;

exports[`Number component renders correctly for negative number 1`] = `
<div>
  <span
    class="number negative"
  >
    -1
  </span>
</div>
`;

exports[`Number component renders correctly for positive number 1`] = `
<div>
  <span
    class="number positive"
  >
    1
  </span>
</div>
`;
